CRM/Contribute/BAO/Contribution.php

```diff
@@ -5418,7 +5418,8 @@ public static function getSalesTaxFinancialAccounts() {
   public static function createProportionalEntry($entityParams, $eftParams) {
     $paid = $entityParams['line_item_amount'] * ($entityParams['trxn_total_amount'] / $entityParams['contribution_total_amount']);
     // Record Entity Financial Trxn
-    $eftParams['amount'] = round($paid, 2);
+    // CRM-20145
+    $eftParams['amount'] = number_format((float)round($paid, 2), 2, '.', '');
```

Comment:
@lcdservices could we make this into a tiny utility function that can be used wherever we are about to store a money value in db? Currently we should always be storing 2 digits after the decimal in core, but cryptocurrencies will need more. @pradpnayak could you ensure we create/use the same utility function throughout work on sales taxes?